src/walk_dirs.rs

```diff
@@ -54,9 +54,10 @@ lazy_static! {
     static ref WORKING_DIR_PATH: PathBuf = current_dir().unwrap();
 }
 
-pub fn discover_fences_and_files(
-    start_path: &str,
+pub fn discover_fences_and_files<'a>(
```

Comment:
What is this lifetime annotation for?  The lifetime 'a isn't used anywhere in the return type or other arguments, and looks like it can be deleted.